lib/rubocop/cop/utils/format_string.rb

```diff
@@ -111,11 +111,9 @@ def max_digit_dollar_num
         private
 
         def parse
-          @source.to_enum(:scan, SEQUENCE).map do
-            FormatSequence.new(
-              Regexp.last_match
-            )
-          end
+          matches = []
+          @source.scan(SEQUENCE) { matches << FormatSequence.new(Regexp.last_match) }
+          matches
```

Comment:
I'd suggest to just merge this change first, without adding TruffleRuby in CI.

I suspect the performance issue (https://github.com/oracle/truffleruby/issues/2081) is related to `parser` and might need changes there so it's unlikely to be solved in 1 day (but I did not investigate further yet).